src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java

```diff
@@ -162,6 +162,17 @@ public boolean updatePassword(Long id, String newPassword) {
         return true;
     }
 
+    @Override
+    public User findByEmailUUID(String emailUUID) {
+        try {
+            return jdbcTemplate.queryForObject(Queries.SQL_SELECT_USER_BY_EMAIL_UUID, userRowMapper, emailUUID);
+        } catch (EmptyResultDataAccessException e) {
+            throw userEntityNotFoundException(e.getMessage(), "email = " + emailUUID);
+        } catch (DataAccessException e) {
+            throw crudException(e.getMessage(), "get", "email = " + emailUUID);
```

Comment:
In some previous PR I requested to pass exception object into yours methods in order to not lost staktrace. Passing e.getMessage() is not enough. Please pass the whole e object